src/main/java/org/junit/runners/parameterized/BlockJUnit4ClassRunnerWithParameters.java

```diff
@@ -73,6 +73,10 @@ private Object createTestUsingFieldInjection() throws Exception {
             int index = annotation.value();
             try {
                 field.set(testClassInstance, parameters[index]);
+            } catch (IllegalAccessException e) {
+                throw new Exception("Cannot set parameter '" + field.getName()
```

Comment:
You can use https://docs.oracle.com/javase/7/docs/api/java/lang/Throwable.html#initCause(java.lang.Throwable) to set the cause.on the `IllegalAccessException`. Can we do that?